mono/metadata/threadpool.c

```diff
@@ -668,10 +668,10 @@ ves_icall_System_Threading_ThreadPool_GetMaxThreadsNative (gint32 *worker_thread
 MonoBoolean
 ves_icall_System_Threading_ThreadPool_SetMinThreadsNative (gint32 worker_threads, gint32 completion_port_threads, MonoError *error)
 {
-	if (completion_port_threads <= 0 || completion_port_threads > threadpool.limit_io_max)
+	if (!mono_lazy_initialize (&status, initialize) || !mono_refcount_tryinc (&threadpool))
 		return FALSE;
 
-	if (!mono_lazy_initialize (&status, initialize) || !mono_refcount_tryinc (&threadpool))
+	if (completion_port_threads <= 0 || completion_port_threads > threadpool.limit_io_max)
 		return FALSE;
```

Comment:
I just realised that, but you also need to decrement the refcount to the `threadpool` with `mono_refcount_dec (&threadpool);` like at https://github.com/mono/mono/pull/8160/files#diff-a3aa8d89566321261a4853d3142ae215R678. This need to be done at https://github.com/mono/mono/pull/8160/files#diff-a3aa8d89566321261a4853d3142ae215R675 and https://github.com/mono/mono/pull/8160/files#diff-a3aa8d89566321261a4853d3142ae215R700